lua/lspconfig/gopls.lua

```diff
@@ -6,7 +6,32 @@ configs.gopls = {
     cmd = { 'gopls' },
     filetypes = { 'go', 'gomod' },
     root_dir = function(fname)
-      return util.root_pattern 'go.work'(fname) or util.root_pattern('go.mod', '.git')(fname)
+      -- First, search for go.work
+      local primary_root = util.root_pattern 'go.work'(fname)
+      if primary_root then
+        return primary_root
+      end
+
+      -- Then, search up the filesystem for go.mod direcrory
+      local go_mod_hierarchy = {}
+      if primary_root then
```

Comment:
duplicate of line 11-13

Reply:
i actually did notice that but as it was in the original commit i was working from i just left it in, ill remove it and do another commit